### PR TITLE
feat: enable bing support

### PR DESCRIPTION
**Pull request by jeeteshchel**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**
- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To clearly categorize this pull request, prefix the pull request title, using one of the following:
  - **feat**: Introduces a new feature or enhancement to the codebase

# Changelog Entry

### Description
- **feat**: Added Bing search integration for Retrieval Augmented Generation (RAG) web searches. This feature allows users to use Bing as a search provider and inject the results directly into the chat experience.

### Added
- Bing search provider integration in the backend (`backend/open_webui/apps/retrieval/web/bing.py`).
- Test data for Bing search results (`backend/open_webui/apps/retrieval/web/testdata/bing.json`).
- Configuration for Bing search in the settings UI (`src/lib/components/admin/Settings/WebSearch.svelte`).

### Changed
- Updated `README.md` to include Bing in the list of supported search providers.
- Modified `backend/open_webui/apps/retrieval/main.py` to integrate Bing search logic.
- Added `DEFAULT_LOCALE` configuration to `main.py`.

### Deprecated
- N/A

### Removed
- N/A

### Fixed
- N/A

### Security
- N/A

### Breaking Changes
- N/A

### Additional Information
- This addition enhances the versatility of the web search feature by including Bing as a potential source for RAG, providing users with more options for web search results.

### Screenshots or Videos
- In comments

### CODE REVIEW

**Comment by jeeteshchel** [comment]
[Bing search feature manual tests.pdf](https://github.com/user-attachments/files/17541659/Bing.search.feature.manual.tests.pdf)

File: backend/open_webui/apps/retrieval/main.py [hunk 1]
```diff
@@ -173,7 +175,7 @@
 app.state.config.SEARCHAPI_ENGINE = SEARCHAPI_ENGINE
 app.state.config.RAG_WEB_SEARCH_RESULT_COUNT = RAG_WEB_SEARCH_RESULT_COUNT
 app.state.config.RAG_WEB_SEARCH_CONCURRENT_REQUESTS = RAG_WEB_SEARCH_CONCURRENT_REQUESTS
-
+app.state.config.DEFAULT_LOCALE = DEFAULT_LOCALE
```
Comment (tjbck) [inline backend/open_webui/apps/retrieval/main.py:178]: `app.state.config.DEFAULT_LOCALE` is not required here.

File: backend/open_webui/apps/retrieval/main.py [hunk 2]
```diff
@@ -1133,6 +1135,13 @@ def search_web(engine: str, query: str) -> list[SearchResult]:
             raise Exception("No SEARCHAPI_API_KEY found in environment variables")
     elif engine == "jina":
         return search_jina(query, app.state.config.RAG_WEB_SEARCH_RESULT_COUNT)
+    elif engine == "bing":
+        return search_bing(
+            app.state.config.DEFAULT_LOCALE,
```
Comment (tjbck) [inline backend/open_webui/apps/retrieval/main.py:1140]: str(DEFAULT_LOCALE)

File: backend/open_webui/apps/retrieval/web/bing.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+log = logging.getLogger(__name__)
+log.setLevel(SRC_LOG_LEVELS["RAG"])
+'''
+Documentation: https://docs.microsoft.com/en-us/bing/search-apis/bing-web-search/overview
+'''
+def search_bing(
+    locale: str, query: str, count: int, filter_list: Optional[list[str]] = None
+) -> list[SearchResult]:
+    endpoint = os.environ['BING_SEARCH_V7_ENDPOINT'] + "/v7.0/search"
+    subscription_key = os.environ['BING_SEARCH_V7_SUBSCRIPTION_KEY']
```
Comment (tjbck) [inline backend/open_webui/apps/retrieval/web/bing.py:20]: Env var should be imported from `env.py`

**Comment by tjbck** [comment]
LGTM, Thanks!

**Pull request merged**